src/libtools/calculator.cc

```diff
@@ -75,5 +76,75 @@ void Calculator::OverwriteDefaultsWithUserInput(const Property &p,
   }
 }
 
+std::vector<std::string> Calculator::GetPropertyChoices(const Property &p) {
+  if (p.hasAttribute("choices")) {
+    std::string att = p.getAttribute<std::string>("choices");
+    Tokenizer tok{att, " ,"};
+    return tok.ToVector();
+  } else {
+    return {""};
+  }
+}
+
+void Calculator::RecursivelyCheckOptions(const Property &p) {
+  for (const Property &prop : p) {
+    if (prop.HasChildren()) {
+      RecursivelyCheckOptions(prop);
+    } else {
+      std::vector<std::string> choices = Calculator::GetPropertyChoices(prop);
+      const std::string &head = choices.front();
+      if (head != "") {
+        try {
+          Calculator::CheckOption(prop, choices);
+        } catch (const std::runtime_error &e) {
+          std::ostringstream oss;
+          oss << "\nThe input value for \"" << prop.name() << "\"";
+          if (choices.size() == 1) {
+            oss << " should be a \"" << head << "\"";
+          } else {
+            oss << " should be one of following values: ";
+            for (const std::string &c : choices) {
+              oss << "\"" << c << "\""
+                  << " ";
+            }
+          }
+          oss << " But \"" << prop.value()
+              << "\" cannot be converted into one.";
+          std::cout << oss.str() << "\n";
+          throw;
+        }
+      }
+    }
+  }
+}
+
+void Calculator::CheckOption(const Property &prop,
+                             const std::vector<std::string> &choices) {
+  const std::string &head = choices.front();
+  std::ostringstream oss;
+
+  if (head == "bool") {
+    prop.as<bool>();
+  } else if (head == "float") {
+    prop.as<double>();
+  } else if (head == "float+") {
+    if (prop.as<double>() < 0.0) {
+      throw std::runtime_error("");
```

Comment:
should these errors not be more verbose? i.e. `option "a" needs a float, your argument "b" cannot be converted to it`

Reply:
That is done above in line 98: https://github.com/votca/tools/blob/choices/src/libtools/calculator.cc#L98
That runtime error is caught there and verbose error is printed. For example, for the [test](https://github.com/votca/tools/blob/choices/src/tests/test_calculator.cc#L95) something like this is printed:
```batch
2: The input value for "option1" should be one of the following values: "foo" "bar" "baz" "qux"  But "boom" cannot be converted into one.
2: The input value for "option2" should be a "float" But "some" cannot be converted into one.
2: The input value for "option3" should be a "int" But "3.14" cannot be converted into one.
2: The input value for "option4" should be a "int+" But "-2" cannot be converted into one.
2: The input value for "option5" should be a "float+" But "-3.14" cannot be converted into one.
```